File: lib/src/models/user/userSummary.dart
import 'package:care_sync/src/models/enums/userRole.dart';

class UserSummary {
  final int id;
  final String email;
  final String? systemEmail;
  final String name;
  final UserRole role;

  UserSummary({
    required this.id,
    required this.email,
    this.systemEmail,
    required this.name,
    required this.role,
  });

  factory UserSummary.fromJson(Map<String, dynamic> json) {
    return UserSummary(
      id: json['id'],
      email: json['email'],
      systemEmail: json['systemEmail'],
      name: json['name'],
      role: UserRole.fromJson(json['role']),
    );
  }
}

File: ios/CareBindrShareExtension/Model/UserSummary.swift
struct UserSummary {
    let id: String
    let name: String
    let email: String

    static func fromJson(_ json: [String: Any]) -> UserSummary? {
        guard let idInt = json["id"] as? Int,
              let name = json["name"] as? String,
              let email = json["email"] as? String else {
            return nil
        }

        return UserSummary(
            id: String(idInt), // ✅ convert Int → String
            name: name,
            email: email
        )
    }
}
